src/render/SDL_palette_sw.c

```diff
@@ -0,0 +1,165 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2022 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "../SDL_internal.h"
+
+/* This is the software implementation of the paletted texture support */
+
+#include "SDL_palette_sw_c.h"
+
+
+SDL_SW_PaletteTexture *
+SDL_SW_CreatePaletteTexture(Uint32 format, int w, int h)
+{
+    SDL_SW_PaletteTexture *swdata;
+
+    switch (format) {
+    /*
+    case SDL_PIXELFORMAT_INDEX1LSB:
+    case SDL_PIXELFORMAT_INDEX1MSB:
+    case SDL_PIXELFORMAT_INDEX4LSB:
+    case SDL_PIXELFORMAT_INDEX4MSB:
+    */
+    case SDL_PIXELFORMAT_INDEX8:
+        break;
+    default:
+        SDL_SetError("Unsupported palette format");
+        return NULL;
+    }
+
+    swdata = (SDL_SW_PaletteTexture *) SDL_calloc(1, sizeof(*swdata));
```

Comment:
I'm not sure about this project's way of coding but a good practice is to use variable declaration with its initialization:
```c
SDL_SW_PaletteTexture *swdata = (SDL_SW_PaletteTexture *) SDL_calloc(1, sizeof(*swdata));
```

This way we reduce the chance of uninitialized pointers being used in the code.

Reply:
Mixed declarations and code is only possible in C99, while SDL uses C89.